Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp

```diff
@@ -427,6 +449,48 @@ void WebKitProtocolHandler::handleGPU(WebKitURISchemeRequest* request)
     }
 #endif
 
+#if PLATFORM(WPE) && ENABLE(WPE_PLATFORM)
+    if (usingWPEPlatformAPI) {
+        std::unique_ptr<PlatformDisplay> platformDisplay;
+#if USE(GBM)
+        UnixFileDescriptor fd;
+        struct gbm_device* device = nullptr;
+        if (const char* node = wpe_display_get_drm_render_node(wpe_display_get_primary())) {
+            fd = UnixFileDescriptor { open(node, O_RDWR | O_CLOEXEC), UnixFileDescriptor::Adopt };
+            if (fd) {
+                device = gbm_create_device(fd.value());
+                if (device)
+                    platformDisplay = PlatformDisplayGBM::create(device);
+            }
+        }
+#endif
+        if (!platformDisplay)
+            platformDisplay = PlatformDisplaySurfaceless::create();
+
+        if (platformDisplay) {
+            auto hardwareAccelerationObject = JSON::Object::create();
+            startTable("Hardware Acceleration Information (Render Process)"_s);
+
+            addTableRow(hardwareAccelerationObject, "Platform"_s, String::fromUTF8(platformDisplay->type() == PlatformDisplay::Type::Surfaceless ? "Surfaceless"_s : "GBM"_s));
+
```

Comment:
At this point, if the GBM `device` is valid, we could also add the chosen DRM driver name and description using the result of calling `drmGetVersion()`.

Reply:
Good point, I plan to add more information, like the buffer format in a follow up patch.

Reply:
Sounds great, thanks!